### PR TITLE
BUG - pd.concat with all Series on axis=1 ignores the `names` argument (Issue: 23490)

### LINKED ISSUE #23490: BUG: pd.concat with all Series on axis=1 ignores the `names` argument.
#### Code Sample, a copy-pastable example if possible

```python
# Example 1
s = pd.Series([1, 2, 3])
df = pd.concat([s, s], axis=1, keys=['a', 'b'], names=['Test'])
df.columns.names  # this gives back FrozenList([None]) instead of FrozenList(['Test'])
    
# Example 2
df = pd.concat([s, s], axis=1, keys=[('a', 1), ('b', 2)], names=['Test1', 'Test2'])
df.columns.names  # this gives back FrozenList([None, None]) instead of FrozenList(['Test1', 'Test2'])

```
#### Problem description

This is a problem because the API allows a user to provide a names argument with these input types and rather than throw an error or take the argument into consideration it is ignored. As far as I can tell, there is no reason for the names argument to be ignored, considering the keys are applied correctly, so I believe this was just an oversight in the `core.reshape.concat._Concatenator._get_concat_axis` method.

#### Proposed Solution

On line 503 of pandas/core/reshape/concat.py the following line:
`return ensure_index(self.keys)`
Should be replaced with something along these lines:

    new_axis = ensure_index(self.keys)
    new_axis.names = self.names
    return new_axis

This solution solves provides the desired output for both examples above and does not break any of the test_concat.py tests.

#### Output of ``pd.show_versions()``
pandas: 0.24.0.dev0

### PR DESCRIPTION
**Pull request by ghost**
- [x] closes #23490
- [x] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

Fix issue described in #23490, and add tests to catch this case.

### CODE REVIEW

File: doc/source/whatsnew/v0.24.0.txt [hunk 1]
```diff
@@ -933,6 +933,7 @@ Reshaping
 ^^^^^^^^^
 
 - Bug in :func:`pandas.concat` when joining resampled DataFrames with timezone aware index (:issue:`13783`)
+- Bug in :func:`pandas.concat` when joining only series the `names` argument of `concat` is no longer ignored (:issue:`23490`)
```
Comment (gfyoung) [inline doc/source/whatsnew/v0.24.0.txt]: series --> ``Series`` (with double backticks)

File: pandas/tests/reshape/test_concat.py [hunk 2]
```diff
@@ -1618,6 +1618,22 @@ def test_concat_series_axis1(self, sort=sort):
         expected = DataFrame({'A': s, 'B': s2})
         assert_frame_equal(result, expected)
 
+        # ensure names argument is not ignored on axis=1
```
Comment (gfyoung) [inline pandas/tests/reshape/test_concat.py]: Reference issue number as a comment.

**Comment by ArtinSarraf** [comment]
@gfyoung - Is there a way to rerun the Travis CI? I get the following error in the logs, and it doesn't seem to be explicitly related to any of my changes.

"No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated"

File: pandas/core/reshape/concat.py [hunk 3]
```diff
@@ -500,7 +500,9 @@ def _get_concat_axis(self):
                 else:
                     return ibase.default_index(len(self.objs))
             else:
-                return ensure_index(self.keys)
+                new_axis = ensure_index(self.keys)
```
Comment (jreback) [inline pandas/core/reshape/concat.py]: use .set_names as this could modify self.keys

File: pandas/tests/reshape/test_concat.py [hunk 4]
```diff
@@ -1618,6 +1618,22 @@ def test_concat_series_axis1(self, sort=sort):
         expected = DataFrame({'A': s, 'B': s2})
         assert_frame_equal(result, expected)
 
+        # ensure names argument is not ignored on axis=1, #23490
```
Comment (jreback) [inline pandas/tests/reshape/test_concat.py]: pls make a new test name

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/reshape/concat.py (hunk 3 above)
Comment (ArtinSarraf) [reply pandas/core/reshape/concat.py]: done

File: pandas/tests/reshape/test_concat.py (hunk 4 above)
Comment (ArtinSarraf) [reply pandas/tests/reshape/test_concat.py]: done

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
great. ping on green.

**Comment by gfyoung** [comment]
Thanks @ArtinSarraf !

**Pull request merged**